### ISSUE TITLE
Strip Quoted reply does not work for all email clients

### ISSUE THREAD

**Issue by pmb399** [osTicket/osTicket#5548]
### Prerequisites
* [**YES** ] Can you reproduce the problem in a fresh installation of the "develop" branch?
* [**NO** ] Do you have any errors in the PHP error log, or javascript console?
* [**YES** ] Did you check the [osTicket forums](https://forum.osticket.com/)?
* [**YES** ] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

### Description

When customer replies are send to osTicket from certain email clients, i.e. Thunderbird, the "strip quoted reply" function does not strip reply emails properly, meaning that the message does not get stripped at all. However, if the customer replies to the same message sent by staff via other email clients, i.e. Android phone, the reply is stripped properly.

The function and setup therefore works and most likely is caused by the customer's email client (Thunderbird). But since we do not have control over our customer's email clients, a fix appears to be necessary on the osTicket level.

### Steps to Reproduce

1. Install fresh osTicket-develop-next instance.
2. Configure email fetching.
3. Send email from Thunderbird (email source attached) - new Ticket will be opened
4. Staff answers ticket.
5. Client sends reply to staff message (via Thunderbird).

E-Mail Header:

[Github Mail Header.pdf](https://github.com/osTicket/osTicket/files/4783017/Github.Mail.Header.pdf)

**Expected behavior:** 
Email answers should have the quotes stripped.

**Actual behavior:**
The reply from the customer does not get stripped when using Thunderbird.
Replys from customer via different email client get stripped properly.

![Github_osTicket_email_Threads](https://user-images.githubusercontent.com/20231767/84705458-a4f7ca80-af18-11ea-857d-8ff4351a192f.png)

### Versions
Latest development version.

Server: Apache
MySQL: 5.7.28
PHP: 7.2.31 - all extension (with exception of APCu working)

**Comment by JediKev** [comment]
@pmb399 

Interesting...I can see where ` -- reply above this line --` is in the email but it's not stripping it there. Is your Reply Separator different than ` -- reply above this line --` (**Admin Panel > Emails > Settings > Reply Separator Tag**)?

Cheers.

**Comment by pmb399** [comment]
Nope, I did not change the default. It's still: -- reply above this line --
And it works properly from my phone...

Cheers!

**Comment by JediKev** [comment]
@pmb399 

Hmm.. I will have to take your example and see if I can replicate this.

Cheers.

**Comment by JediKev** [comment]
@pmb399 

Can you provide a full raw email (including headers) that is not being stripped so I can attempt to replicate? I used the content in the PDF you provided in your OP but it's malformed and unusable.

Cheers.

**Comment by pmb399** [comment]
@JediKev 
Yup. Should I just upload it here? Or do you have a more secure spot.

**Comment by JediKev** [comment]
@pmb399 

You can upload it here, just remember to remove/sensor (`XXX`) sensitive data beforehand.

Cheers.

**Comment by pmb399** [comment]
@JediKev 

Here is the new raw header. It opens fine in my email program when loading the source code as eml. I think I just did not properly censor it before which is why it was malformatted. My apologies.

[EMail.eml.pdf](https://github.com/osTicket/osTicket/files/4783171/EMail.eml.pdf)

**Comment by pmb399** [comment]
@JediKev 
Actually, I am not sure if it likes being written to pdf, so here is just a txt version. Thanks.

[EMail_new.TXT](https://github.com/osTicket/osTicket/files/4783215/EMail_new.TXT)

**Comment by JediKev** [comment]
@pmb399 

Okay, so after some testing, the reason it's not being stripped is because when the email is parsed in the system `-- reply above this line --` (within the content) turns into:
```
--
        reply above this line --
```
(notice the new line and overload of spaces)

I have no idea why it looks like this when parsed. 🤔 I will have to discuss this with the team and get back to you.

Cheers.

**Comment by JediKev** [comment]
@pmb399 

I see the issue now, missed it the first time. In the `text/html` section, the Reply Separator is already malformed. So the question would be, does the original email look like that? (original email being the one that's quoted)

Cheers.

**Comment by pmb399** [comment]
@JediKev 
Good catch with the initial email. That one is base64 encoded. As soon as I open it in Thunderbird, the reply separator looks like you described.

This is the html code from Thunderbird from the original email:
![Screenshot from 2020-06-17 17-51-38](https://user-images.githubusercontent.com/20231767/84962190-7a964080-b0c3-11ea-8621-19629cda046e.png)

It looks properly when I encode the email via terminal.
![Screenshot from 2020-06-17 17-58-38](https://user-images.githubusercontent.com/20231767/84962486-348dac80-b0c4-11ea-9f09-32c023e6ab6c.png)

Now, the easiest work around would be to change the reply separator to a string without any empty spaces, i.e. `--reply_above_this_line--`. I just tried this and it works as expected with all my email clients!

However, since my production system runs with the default separator tag (with too many open tickets), I am a little reluctant to change the separator, unless I could hard-code in the default separator tag with an `or` statement or something among those lines...

Cheers!

**Comment by JediKev** [comment]
@pmb399 

In theory you could get away with using `reply above this line --` (without the beginning dashes) and that should match the old ones and new ones.

Cheers.

**Comment by pmb399** [comment]
@JediKev 
Unfortunately, this only works for the "old" separator tag `-- reply above this line --`. The parsed string then matches the criteria from any email client. It, however, does not work when replying to emails with the new separator tag you suggested without the leading dashes `reply above this line --`

This is how Thunderbird renders the html code with the new tag:

![Screenshot from 2020-06-17 18-32-55](https://user-images.githubusercontent.com/20231767/84964444-5ccbda00-b0c9-11ea-878a-386a4ddc545f.png)

As such, the "reply" is not in the same line with the remaining tag and the entire email gets read.

This might just be a Thunderbird issue. However, I assume changing the string to not include white spaces might be the easiest/safest fix. 

Any chance adding a second separator to avoid loading in junk from previous tickets is doable as I suggested in my last post? Or is there a way to differentiate between the separator tag that is being _sent_ versus the separator tag that is searched for when the message is _parsed_? Then you could send `-- -- reply above this line` and look for `-- reply above this line --`

Cheers!

**Comment by mrkylegp** [comment]
@JediKev I can also confirm this is an issue. In my case I've tested replying via Thunderbird and it doesn't strip everything below the separator tag if I modify the separator tag. If I use exactly -- reply above this line -- in the email template and set the reply separator tag to be exactly that as well, it works fine. If I try to use something like -- reply above this line or in the ticket system -- in both the reply separator tag setting field and the email templates, the quoted reply doesn't get stripped.

**Comment by thompol** [comment]
Same problem here. Changing the separator to one without spaces (e.g. `--reply_above_this_line--`) works, like suggested here.

**Comment by mrkylegp** [comment]
I'm now on v1.17 and this still doesn't appear to be working.

I've used exactly -- reply above this line -- and I've tried putting that text at the very top of my email template.

--reply_above_this_line-- (no spaces) does in fact work as discussed above.

**Comment by Jieiku** [comment]
I changed it to 

`--Original-Message--`

I am also attempting to put one at the start and end of the message, because often people will also reply after the original message (they don't always follow directions) This results in a blank reply in the osTicket admin panel ticket info.

I think when the reply is an empty reply, that it should disable the Strip Quoted reply for that response. Anytime somebody replys to our email in a goofy way (either at the bottom, and sometimes right in the middle of the original message! ) then we simply see an empty response from that person, it makes Strip Quoted reply an almost useless feature as it is now.

Here is that thread:
https://github.com/osTicket/osTicket/issues/6470

Here is where I am trying to improve it:
https://github.com/Jieiku/osTicket/commit/0a6dea82c020cd94cc3406b254328df9f39477a9

I also need to add some logic that checks if the response will be empty after performing the strip quoted reply, and if so just send the entire reply without stripping.

EDIT: I just found part of the problem... roundcube's default is to put replys below the quoted text, while thunderbird defaults to replying above (which is good)

to fix roundcube you can set it to 1 (default is 0):

```php
// When replying:
// -1 - don't cite the original message
// 0  - place cursor below the original message
// 1  - place cursor above original message (top posting)
// 2  - place cursor above original message (top posting), but do not indent the quote
$config['reply_mode'] = 1;
```
